src/Data/Nat/DivMod.agda

```diff
@@ -115,6 +115,26 @@ m<[1+n%d]⇒m≤[n%d] {m} n (suc d-1) = k<1+a[modₕ]n⇒k≤a[modₕ]n 0 m n d-
   (m % d +  n % d + (n / d) * d)  % d ≡⟨ [m+kn]%n≡m%n (m % d + n % d) (n / d) d-1 ⟩
   (m % d +  n % d)                % d ∎
 
+%-distribˡ-* : ∀ m n d {≢0} → ((m * n) % d) {≢0} ≡ (((m % d) {≢0} * (n % d) {≢0}) % d) {≢0}
+%-distribˡ-* m n d@(suc d-1) = begin-equality
+  (m * n)                                             % d ≡⟨ cong (λ h → (h * n) % d) (m≡m%n+[m/n]*n m d-1) ⟩
+  ((m′ + k * d) * n)                                  % d ≡⟨ cong (λ h → h % d) (*-distribʳ-+ n m′ (k * d)) ⟩
+  (m′ * n + (k * d) * n)                              % d ≡⟨ cong (λ h → (m′ * h + (k * d) * h) % d) (m≡m%n+[m/n]*n n d-1) ⟩
+  (m′ * (n′ + j * d) + (k * d) * (n′ + j * d))        % d ≡⟨ cong (λ h → (h + (k * d) * (n′ + j * d)) % d) (*-distribˡ-+ m′ n′ (j * d)) ⟩
```

Comment:
From here onwards should be solvable automatically using the new `Data.Nat.Tactic.RingSolver` module. Although you have may have to expand out the variables in the `where` block. See `README.Tactic.RingSolver` for details on how to use.

Reply:
I added lemma. I could not get `solve` to work - only `solve-∀`.
I'd note that it doesn't show up in https://agda.github.io/agda-stdlib/README.html.

Reply:
Thanks, filed as #1083.